### PR TITLE
Update max_step calculation in LR Scheduler

### PR DESCRIPTION
**Pull request by blisc**

### CODE REVIEW

File: nemo/core/optim/lr_scheduler.py [hunk 1]
```diff
@@ -501,33 +503,39 @@ def prepare_lr_scheduler(
...
-        iters_per_batch = scheduler_config.get('iters_per_batch')
+        max_epochs = scheduler_config.get('t_max_epochs')
+        accumulate_grad_batches = scheduler_config.get('t_accumulate_grad_batches')
+        num_workers = scheduler_config.get('t_num_workers')
 
         # Compute effective num max_steps
         num_samples = len(train_dataloader.dataset)
         batch_size = train_dataloader.batch_size
-        max_steps = round(num_samples * iters_per_batch / float(batch_size))
+        drop_last = train_dataloader.batch_size
```
Comment (titu1994) [inline nemo/core/optim/lr_scheduler.py:534]: Drop_last = train dataloader batch size ?

File: tests/core/test_optimizers_schedulers.py [hunk 2]
```diff
@@ -631,3 +705,47 @@ def test_InverseSquareRootAnnealing(self):
...
+            max_steps = optim.lr_scheduler.compute_max_steps(
+                max_epochs, accumulate_grad_batches, num_processes, dataset_len, batch_size, drop_last
+            )
+            model = ExampleModel(batch_size, dataset_len, drop_last, max_steps)
+            trainer.callbacks.append(Callback())
+            trainer.fit(model)
+
+        # Test drop_last = False, all other parameters free
+        for _ in range(3):
+            drop_last = False
```
Comment (titu1994) [inline tests/core/test_optimizers_schedulers.py:731]: Maybe fix seeds for tests so they aren't flaky? Doubt it's an issue though.

**Review by titu1994 — APPROVED** [review]
LGTM. Thanks for this !

File: nemo/core/optim/lr_scheduler.py (hunk 1 above)
Comment (blisc) [reply nemo/core/optim/lr_scheduler.py:534]: That is a good catch

**Review by okuchaiev — APPROVED** [review]
looks good, thank you!
